Update subsection and unit screen for gated content

miankhalid opened the pull request:
### Description

[LEARNER-6640](https://openedx.atlassian.net/browse/LEARNER-6640)

- Blocks API has been bumped from v1 to v2.
- CourseUnitMobileNotSupportedFragment class and its layout have been
refactored to use data binding.

Subsection Screen | Unit Screen
--- | ---
![Subscection](https://user-images.githubusercontent.com/1710804/60443004-b6769280-9c33-11e9-8fd6-b0421f5b4fb4.png) | ![not_available_lock](https://user-images.githubusercontent.com/1710804/60443018-be363700-9c33-11e9-8ed9-1225da04998a.png)

miankhalid commented:
@farhan-arshad-dev plz review.

farhan-arshad-dev commented on OpenEdXMobile/res/layout/fragment_course_unit_grade.xml:
```diff
@@ -1,55 +1,60 @@
-        android:layout_height="wrap_content"
-        android:layout_marginBottom="20dp"
-        android:text="@string/assessment_view_on_web" />
+        <Button
```
Are we need to show the `View on Web` button? because the use can't access the course section on web as well  that have `authorization_denial_reason` key.

miankhalid replied:
Yes, we need it, cuz on the web the user can then pay for the verified track and access the specific component.

farhan-arshad-dev commented:
@miankhalid Travis-ci build is also being failed due to blocks api `v2` and related model changes. We also need to update the mock response for `blocks` API.

miankhalid commented:
@farhan-arshad-dev ready for another pass.

**Note:** I've fixed your review points and added the following 2 things too:

- Pick API version for Blocks API from configs
- Define enum for Authorization Denial Reason

p.s. the tests will still fail, I'll fix those issues in the morning.
